File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    
    int a, b, c;
    int max_side;

    while (true) {
        cin >> a >> b >> c;

        if (a == 0 && b == 0 && c == 0)
            break;

        if (a > b && a > c)
            max_side = a;
        else if (b > a && b > c)
            max_side = b;
        else
            max_side = c;

        if (max_side >= a + b + c - max_side)
            cout << "Invalid" << "\n";
        else if (a == b && b == c)
            cout << "Equilateral" << "\n";
        else if (a == b || b == c || c == a)
            cout << "Isosceles" << "\n";
        else
            cout << "Scalene" << "\n";
    }
    
    return 0;
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys

while True:
    side = list(map(int, sys.stdin.readline().split()))

    if side[0] == 0 and side[1] == 0 and side[2] == 0:
        break;

    if max(side) >= sum(side) - max(side):
        sys.stdout.write("Invalid\n")
    elif side[0] == side[1] == side[2]:
        sys.stdout.write("Equilateral\n")
    elif side[0] == side[1] or side[1] == side[2] or side[2] == side[0]:
        sys.stdout.write("Isosceles\n")
    else:
        sys.stdout.write("Scalene\n")
        